src/iosource/PktSrc.cc

```diff
@@ -369,3 +330,22 @@ bool PktSrc::GetCurrentPacket(const Packet** pkt)
 	*pkt = &current_packet;
 	return true;
 	}
+
+double PktSrc::GetNextTimeout()
```

Comment:
This is nice how it captures all the 3 cases through a single method.

Reply:
👍